chef/data_bags/crowbar/migrate/provisioner/022_upgrade_autoyast_repos.rb

```diff
@@ -12,10 +18,16 @@ def upgrade ta, td, a, d
 end
 
 def downgrade ta, td, a, d
-  if !a['suse']['autoyast']['repos']['suse-11.3'].nil? && !a['suse']['autoyast']['repos']['suse-11.3'].empty?
-    a['suse']['autoyast']['repos']      = a['suse']['autoyast']['repos']['suse-11.3']
-  elsif !a['suse']['autoyast'].nil?
-    a['suse'].delete('autoyast')
+  # The 'suse' hierarchy is optional in the proposal template
+  if !a['suse'].nil?
+    if !a['suse']['autoyast'].nil? &&
+          !a['suse']['autoyast']['repos'].nil? &&
+          !a['suse']['autoyast']['repos']['suse-11.3'].nil? &&
+          !a['suse']['autoyast']['repos']['suse-11.3'].empty?
+      a['suse']['autoyast']['repos'] = a['suse']['autoyast']['repos']['suse-11.3']
+    else
+      a.delete('suse')
```

Comment:
We did not introduce the ``'suse'`` section with the previous commits, only the structure under ``['autoyast']['repos']`` was changed. ...